retrieval/transform_test.go

```diff
@@ -626,6 +626,88 @@ func TestSampleBuilder(t *testing.T) {
 				},
 			},
 		},
+		// Metrics with the _total suffix should be preserved if metadata can be found for the original metric name.
+		{
+			series: seriesMap{
+				1: labels.FromStrings("job", "job1", "instance", "instance1", "a", "1", "__name__", "metric1_total"),
+			},
+			targets: targetMap{
+				"job1/instance1": &targets.Target{
+					Labels:           promlabels.FromStrings("job", "job1", "instance", "instance1"),
+					DiscoveredLabels: promlabels.FromStrings("__resource_a", "resource2_a"),
+				},
+			},
+			metadata: metadataMap{
+				"job1/instance1/metric1_total": &scrape.MetricMetadata{Type: textparse.MetricTypeGauge, Metric: "metric1_total"},
```

Comment:
Conventionally `_total` is used with counter type, not gauge type. See https://prometheus.io/docs/practices/naming/
Here is the quote from the page: "... Note that an `accumulating count` has `total` as a suffix..."

Moreover, my understanding is that gauge metrics with `_total` suffix shouldn't use metadata from `metric1` either, as [gauge type does not use _child_samples](https://github.com/prometheus/client_python/blob/ae86f1ce2884f64765868c2150aa630f2d5516e0/prometheus_client/metrics.py#L378-L379) while [counter type has _child_samples with `_total` suffix](https://github.com/prometheus/client_python/blob/ae86f1ce2884f64765868c2150aa630f2d5516e0/prometheus_client/metrics.py#L255-L259)

Reply:
Awesome feedback, thanks! Updated